integrations/linear/src/webhooks.ts

```diff
@@ -0,0 +1,341 @@
+import {
+  ExternalSource,
+  ExternalSourceTrigger,
+  HandlerEvent,
+  IntegrationTaskKey,
+  Logger,
+} from "@trigger.dev/sdk";
+import {
+  LinearWebhooks,
+  LINEAR_WEBHOOK_SIGNATURE_HEADER,
+  LINEAR_WEBHOOK_TS_FIELD,
+  WebhookPayload,
+  DeletePayload,
+  Webhook,
+} from "@linear/sdk";
+import { z } from "zod";
+import * as events from "./events";
+import { Linear, LinearRunTask } from "./index";
+import {
+  WebhookCreateInput,
+  WebhookUpdateInput,
+  WebhooksQueryVariables,
+} from "@linear/sdk/dist/_generated_documents";
+import { WebhookPayloadSchema } from "./schemas";
+
+export const WebhookResourceTypeSchema = z.union([
+  z.literal("Attachment"),
+  z.literal("Comment"),
+  z.literal("Cycle"),
+  z.literal("Issue"),
+  z.literal("IssueLabel"),
+  z.literal("Project"),
+  z.literal("ProjectUpdate"),
+  z.literal("Reaction"),
+]);
+export type WebhookResourceType = z.infer<typeof WebhookResourceTypeSchema>;
+
+export const WebhookActionTypeSchema = z.union([
+  z.literal("create"),
+  z.literal("remove"),
+  z.literal("update"),
+]);
+export type WebhookActionType = z.infer<typeof WebhookActionTypeSchema>;
+
+type DeleteWebhookParams = {
+  id: string;
+};
+
+type UpdateWebhookParams = {
+  id: string;
+  input: WebhookUpdateInput;
+};
+
+// TODO: types
+const withoutFunctions = <T>(obj: T): T => {
+  return JSON.parse(JSON.stringify(obj), (key, value) => {
+    if (typeof value === "function") {
+      return undefined;
+    }
+    return value;
+  });
+};
```

Comment:
This will likely have to be used in many places to get serializable output. Plan is to create a useful generic to go along with this.

Reason: No easy way to get clean entity outputs from the Linear SDK. This goes for all endpoints. I would like to avoid using raw GraphQL queries.

Reply:
Totally agree on avoiding GraphQL queries wherever possible.

We could add a function like this (I think you've updated it) to packages/integration-kit? There are some other useful things in there